src/test/java/com/streisky/virtualshowcasebackend/controller/login/LoginControllerTest.java

```diff
@@ -0,0 +1,133 @@
+                .perform(
+                        MockMvcRequestBuilders
+                                .post(LOGIN_URL)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(accountDTORequestJson.write(AccountTestsConstants.ACCOUNT_DTO).getJson())
+                )
+                .andReturn()
+                .getResponse();
+
+        Assertions.assertEquals(HttpStatus.OK.value(), response.getStatus());
+        Assertions.assertTrue(response.getContentAsString().contains("\"token\""));
+    }
+
+    @Test
+    public void shouldReturnUnauthorized_whenBothCredentialsAreIncorrect() throws Exception {
+        AccountDTO accountDTO = new AccountDTO(AccountTestsConstants.INVALID_LOGIN, AccountTestsConstants.INVALID_PASSWORD);
+        shouldReturnUnauthorized(accountDTO);
+    }
+
+    @Test
+    public void shouldReturnUnauthorized_whenLoginAreIncorrect() throws Exception {
+        AccountDTO accountDTO = new AccountDTO(AccountTestsConstants.INVALID_LOGIN, AccountTestsConstants.PASSWORD);
+        shouldReturnUnauthorized(accountDTO);
+    }
+
+    @Test
+    public void shouldReturnUnauthorized_whenPasswordAreIncorrect() throws Exception {
+        AccountDTO accountDTO = new AccountDTO(AccountTestsConstants.LOGIN, AccountTestsConstants.INVALID_PASSWORD);
+        shouldReturnUnauthorized(accountDTO);
+    }
+
+    private void shouldReturnUnauthorized(AccountDTO accountDTO) throws Exception {
+        accountRepository.save(AccountTestsConstants.ACCOUNT);
+
+        MockHttpServletResponse response = mockMvc
+                .perform(
+                        MockMvcRequestBuilders
+                                .post(LOGIN_URL)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(accountDTORequestJson.write(accountDTO).getJson())
+                )
+                .andReturn()
+                .getResponse();
+
+        Assertions.assertEquals(HttpStatus.UNAUTHORIZED.value(), response.getStatus());
+    }
+
+    @Test
+    public void shouldReturnBadRequest_WhenLoginIsNull() throws Exception {
+        AccountDTO accountDTO = new AccountDTO(null, AccountTestsConstants.PASSWORD);
+        shouldReturnBadRequest(accountDTO, "login");
+    }
+
+    @Test
+    public void shouldReturnBadRequest_WhenPasswordIsNull() throws Exception {
+        AccountDTO accountDTO = new AccountDTO(AccountTestsConstants.LOGIN, null);
+        shouldReturnBadRequest(accountDTO, "password");
+    }
+
+    @Test
+    public void shouldReturnBadRequest_WhenLoginIsEmpty() throws Exception {
+        AccountDTO accountDTO = new AccountDTO("", AccountTestsConstants.PASSWORD);
+        shouldReturnBadRequest(accountDTO, "login");
+    }
+
+    @Test
+    public void shouldReturnBadRequest_WhenPasswordIsEmpty() throws Exception {
+        AccountDTO accountDTO = new AccountDTO(AccountTestsConstants.LOGIN, "");
```

Comment:
A suggestion would be to create entire object on the mocks file and avoid having to use the primitive types here.
For example: 

accountDTOWithoutPassword

Reply:
In this case, I would have to create a const to each situation. One to without passord, one to without login, one to password is null and one to login is null?

Reply:
Only if the object was reused a lot. Otherwise, I think it would be ok to use literal values on the test class, that is very common practice.

Reply:
For example. If for each controller class you create a constants file, I think it would make more sense to just keep the constants on the test class.